qiskit_experiments/framework/base_analysis.py

```diff
@@ -153,30 +153,48 @@ def run(
         if not replace_results and _requires_copy(experiment_data):
             experiment_data = experiment_data.copy()
 
-        experiment_components = self._get_experiment_components(experiment_data)
-
         # Set Analysis options
         if not options:
             analysis = self
         else:
             analysis = self.copy()
             analysis.set_options(**options)
 
-        def run_analysis(expdata):
+        def run_analysis(expdata: ExperimentData):
             # Clearing previous analysis data
             experiment_data._clear_results()
-            # making new analysis
+            experiment_components = self._get_experiment_components(experiment_data)
+
+            # Making new analysis
             results, figures = analysis._run_analysis(expdata)
-            # Add components
-            analysis_results = [
-                analysis._format_analysis_result(
-                    result, expdata.experiment_id, experiment_components
-                )
-                for result in results
-            ]
-            # Update experiment data with analysis results
-            if analysis_results:
-                expdata.add_analysis_results(analysis_results)
+
+            if results:
+                for result in results:
+                    supplementary = result.extra
```

Comment:
This is intentional and composite analysis requires this logic to update entry experiment name. Since we should return `AnalysisResultData` from the child experiment data, it [updates experiment value through extra](https://github.com/Qiskit/qiskit-experiments/pull/1133/files#diff-68444ee92b9659754a66f17cf15e60e985b92fefb414e3234cf4dd499a4b559aR372) field. Alternatively, we can completely remove child data generation in the composite analysis, and generate them on the fly when user access `ExperimentData.child_data` for backward compatibility. This drastically simplifies the code stack, but should be done in another PR because I don't want to add multiple logical changes in a single PR.